rllib/algorithms/impala/impala.py

```diff
@@ -932,18 +946,24 @@ def learn_on_processed_samples(self) -> ResultDict:
                 num_iters=self.config.num_sgd_iter,
                 minibatch_size=self.config.minibatch_size,
             )
+        # Nothing on the queue -> Don't send requests to learner group.
         else:
             lg_results = None
 
         if lg_results:
-            self._counters[NUM_ENV_STEPS_TRAINED] += lg_results[ALL_MODULES][
+            self._counters[NUM_ENV_STEPS_TRAINED] += lg_results[ALL_MODULES].pop(
                 NUM_ENV_STEPS_TRAINED
-            ]
-            self._counters[NUM_AGENT_STEPS_TRAINED] += lg_results[ALL_MODULES][
+            )
+            self._counters[NUM_AGENT_STEPS_TRAINED] += lg_results[ALL_MODULES].pop(
                 NUM_AGENT_STEPS_TRAINED
+            )
+            lg_queue_stats = self.learner_group.get_in_queue_stats()
+            self._counters["learner_group_queue_size"] = lg_queue_stats[
```

Comment:
hmm why not instead `self._counters.update(lg_queue_stats)` unless you want to be very specific about the names.

Reply:
Great catch! Fixed.